File: Isomorphic Strings - GFG/isomorphic-strings.cpp
//{ Driver Code Starts
// C++ program to check if two strings are isomorphic


#include<bits/stdc++.h>


using namespace std;
#define MAX_CHARS 256

// } Driver Code Ends
class Solution
{
    public:
    //Function to check if two strings are isomorphic.
    bool areIsomorphic(string s, string t)
    {
        
        // Your code here
        int res1[256]={0}, res2[256]={0};
        
        if(s.size()!=t.size())
            return false;
        for(int i = 0 ; i < s.size() ; i++){
            if(res1[s[i]]!=res2[t[i]])
                return false;
            res1[s[i]] = i+1;
            res2[t[i]] = i+1;
            
        }
        
        
       return true; 
    }
};

//{ Driver Code Starts.

// Driver program
int main()
{
    int t;
    cin>>t;
    string s1,s2;
    while (t--) {
        cin>>s1;
        cin>>s2;
        Solution obj;
        cout<<obj.areIsomorphic(s1,s2)<<endl;
    }
    
    return 0;
}
// } Driver Code Ends

File: Isomorphic Strings - GFG/isomorphic-strings.py
#User function Template for python3

class Solution:
    
    #Function to check if two strings are isomorphic.
    def areIsomorphic(self,str1,str2):
        res1 = []
        res2 = []
        for i in str1:
            res1.append(str1.index(i))
        for j in str2:
            res2.append(str2.index(j))
        if res2==res1:
            return True
        return False
            


#{ 
 # Driver Code Starts
#Initial Template for Python 3

import atexit
import io
import sys
from collections import defaultdict

_INPUT_LINES = sys.stdin.read().splitlines()
input = iter(_INPUT_LINES).__next__
_OUTPUT_BUFFER = io.StringIO()
sys.stdout = _OUTPUT_BUFFER

@atexit.register

def write():
    sys.__stdout__.write(_OUTPUT_BUFFER.getvalue())

if __name__=='__main__':
    t = int(input())
    for i in range(t):
        s=str(input())
        p=str(input())
        ob = Solution()
        if(ob.areIsomorphic(s,p)):
            print(1)
        else:
            print(0)
# } Driver Code Ends